src/main/java/org/wahlzeit_revisited/repository/AbstractRepository.java

```diff
@@ -0,0 +1,121 @@
+package org.wahlzeit_revisited.repository;
+
+import org.wahlzeit.services.SysLog;
+import org.wahlzeit_revisited.database.DatabaseConnection;
+import org.wahlzeit_revisited.database.SessionManager;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Optional;
+
+public abstract class AbstractRepository<T extends Persistent> implements Repository<T> {
```

Comment:
I like that repository pattern since it seems to me separating concerns better.. Let's ask Dirk whether he has a strong opinion there

Reply:
How is the Repository pattern (had to read up on it) different from the Manager pattern? Both are supposed to decouple domain objects from their persistence and retrieval? Repository = only load and save? Manager = Repo + search functionality? In general, I find it hard to follow on the code level if we need a clean design for students Do you have some proposed class diagrams for the refactorings?

Reply:
The key difference is, that we do no longer cache Objects in the main memory and therefore do not perform any actions on them

A Repository also implements the search functionality, but it focuses more on the database/SQL side.
Search and retrievals in a Repository are performed over an specific SQL-Query, followed by the POJO instantiation. To test the non-trivial SQL-Queries, I added an Integration-tests for each.
As I understood the `Manager`-approach, it initially fetched everything from the Database with an trivial/generall SQL query, and then performing all the filtering and searching on the Java-Objects. 

I chose this pattern, at it the the preferred JPA pattern of Spring boot: [reference link](https://www.baeldung.com/spring-data-repositories)

An brief outline for the whole JavaEE approach:
- `*Resource` class uses the JavaEE annotations, to define our api, so routing, POST; GET, etc. Also the AccessRights are handled here over annotations. Each `Resource`-Method calls it's according `Service`-Method.
- *Service` is quite similar the to previous `Handlers`, where the business logic lives. Each service has access to it's depending `Repository`(s) and `Factory`(s), which are basically what previously a `Manager` did, but without the main memory caching. I try to reimplement that part as close to the original codebase as possible.
- The rest of the application will stay untouched, as the deeper business logic has nothing todo with JavaEE